Tasks/AzureFunctionOnKubernetesV0/src/deploy.ts

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+import * as yaml from 'js-yaml';
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as tr from "azure-pipelines-task-lib/toolrunner";
+import { CommandHelper } from './utils/commandHelper';
+import { DockerConnection } from './dockerConnection';
+import { Kubectl, Resource } from 'kubernetes-common-v2/kubectl-object-model';
+import * as FileHelper from './utils/fileHelper';
+import * as KubernetesConstants from 'kubernetes-common-v2/kubernetesconstants';
+import * as KubernetesManifestUtility from 'kubernetes-common-v2/kubernetesmanifestutility';
+import * as CommonUtils from 'kubernetes-common-v2/utility';
+
+const secretName = tl.getInput('secretName');
+const appName = tl.getInput('appName', true);
+const namespace = getKubernetesNamespace();
+let imageName = tl.getInput('imageName');
+const registry = tl.getInput('registry');
+const args = tl.getInput('arguments');
+const funcRootDir = tl.getInput('functionRootDirectory', true);
+const waitForStability = tl.getBoolInput('waitForStability');
+const isDryRun = (args && args.includes('--dry-run')) ? true : false;
+
+let pullSecretName = `${appName}-imagepullsecret`;
+
+export async function deploy(commandHelper: CommandHelper, dockerConnection: DockerConnection) {
+    if (!isDryRun) {
+        // create pull secret if it is not dry-run
+        createImagePullSecret(commandHelper);
```

Comment:
docker login is optional if I'm not wrong; shouldn't you be doing this only if the dockerRegistryConnection is non-empty?

Reply:
You could be accidentally be using a different login otherwise. 
Imagine I have set up a private runner and logged in to docker using my credentials on that machine; you'd end up writing that `DOCKER_CONFIG` instead, right?

Reply:
I believe `DOCKER_CONFIG` set on machine is not available on the task. It is available if it is set in a previous task using `tl.setVariable` (Docker login). I intended to support Docker login with this one.
But based on the discussion with Shashank and Vikranth, we will be making Docker SC a mandatory input. Will change the code accordingly.